### PR TITLE
feat: redirect waypoint plugins to integrations

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Adds redirects from `/waypoint/plugins` to `/waypoint/integrations`.

🚧 **TODO**: Determine whether we also want to remove `/waypoint/plugins` content, and links to Waypoint plugins from within Dev Dot (such as in the top nav). If so, should probably follow this PR up pretty quickly with that work. We'd remove `src/pages/waypoint/plugins`, and the top nav link, and there are probably some other stray links we'd want to update. As well, we should ensure the Waypoint team knows that content in `website/content/plugins` will no longer be rendered to Dev Dot.

## 🛠️ How

Adds a redirect from `/waypoint/plugins/:slug` to `/waypoint/integrations/BrandonRomano/:slug`, as well as a simple redirect from`/waypoint/plugins` to `/waypoint/integrations`.

 🚧 **TODO**: likely want to update `BrandonRomano` to be `HashiCorp` in URLs. Or `hashicorp`? Not sure on casing. This change probably makes sense to do _after_ moving integrations over so the API reflects this change - which I think would be captured by 🎟 [Migrate `BrandonRomano/waypoint` fork to `HashiCorp/waypoint` branch](https://app.asana.com/0/1203792701577283/1203868826798341).

## 🧪 Testing

Visit all of the following `/waypoint/plugins` links, and confirm they redirect as expected, to pages that exist:

- [/waypoint/plugins][/waypoint/plugins]
- [/waypoint/plugins/aws-ec2][/waypoint/plugins/aws-ec2]
- [/waypoint/plugins/aws-ecs][/waypoint/plugins/aws-ecs]
- [/waypoint/plugins/aws-lambda][/waypoint/plugins/aws-lambda]
- [/waypoint/plugins/aws-ssm][/waypoint/plugins/aws-ssm]
- [/waypoint/plugins/azure-container-instance][/waypoint/plugins/azure-container-instance]
- [/waypoint/plugins/consul][/waypoint/plugins/consul]
- [/waypoint/plugins/docker][/waypoint/plugins/docker]
- [/waypoint/plugins/exec][/waypoint/plugins/exec]
- [/waypoint/plugins/google-cloud-run][/waypoint/plugins/google-cloud-run]
- [/waypoint/plugins/helm][/waypoint/plugins/helm]
- [/waypoint/plugins/kubernetes][/waypoint/plugins/kubernetes]
- [/waypoint/plugins/nomad][/waypoint/plugins/nomad]
- [/waypoint/plugins/pack][/waypoint/plugins/pack]
- [/waypoint/plugins/packer][/waypoint/plugins/packer]
    - 🚧 **TODO** - this integrations page 404s. Is this a content issue? Do we need a specific redirect here? It seems like [we have a Packer `builtin` plugin-slash-integration](https://github.com/BrandonRomano/waypoint/tree/main/builtin/packer), so this might just be a case of it not yet being extracted by the Integrations API.
- [/waypoint/plugins/terraform-cloud][/waypoint/plugins/terraform-cloud]
- [/waypoint/plugins/vault][/waypoint/plugins/vault]

## 💭 Anything else?

Not at the moment!

[task]: https://app.asana.com/0/1203792701577283/1203812579407375/f
[preview]: https://dev-portal-git-zsredirect-waypoint-plugins-hashicorp.vercel.app/
[/waypoint/plugins]: https://dev-portal-git-zsredirect-waypoint-plugins-hashicorp.vercel.app/waypoint/plugins
[/waypoint/plugins/aws-ec2]: https://dev-portal-git-zsredirect-waypoint-plugins-hashicorp.vercel.app/waypoint/plugins/aws-ec2
[/waypoint/plugins/aws-ecs]: https://dev-portal-git-zsredirect-waypoint-plugins-hashicorp.vercel.app/waypoint/plugins/aws-ecs
[/waypoint/plugins/aws-lambda]: https://dev-portal-git-zsredirect-waypoint-plugins-hashicorp.vercel.app/waypoint/plugins/aws-lambda
[/waypoint/plugins/aws-ssm]: https://dev-portal-git-zsredirect-waypoint-plugins-hashicorp.vercel.app/waypoint/plugins/aws-ssm
[/waypoint/plugins/azure-container-instance]: https://dev-portal-git-zsredirect-waypoint-plugins-hashicorp.vercel.app/waypoint/plugins/azure-container-instance
[/waypoint/plugins/consul]: https://dev-portal-git-zsredirect-waypoint-plugins-hashicorp.vercel.app/waypoint/plugins/consul
[/waypoint/plugins/docker]: https://dev-portal-git-zsredirect-waypoint-plugins-hashicorp.vercel.app/waypoint/plugins/docker
[/waypoint/plugins/exec]: https://dev-portal-git-zsredirect-waypoint-plugins-hashicorp.vercel.app/waypoint/plugins/exec
[/waypoint/plugins/google-cloud-run]: https://dev-portal-git-zsredirect-waypoint-plugins-hashicorp.vercel.app/waypoint/plugins/google-cloud-run
[/waypoint/plugins/helm]: https://dev-portal-git-zsredirect-waypoint-plugins-hashicorp.vercel.app/waypoint/plugins/helm
[/waypoint/plugins/kubernetes]: https://dev-portal-git-zsredirect-waypoint-plugins-hashicorp.vercel.app/waypoint/plugins/kubernetes
[/waypoint/plugins/nomad]: https://dev-portal-git-zsredirect-waypoint-plugins-hashicorp.vercel.app/waypoint/plugins/nomad
[/waypoint/plugins/pack]: https://dev-portal-git-zsredirect-waypoint-plugins-hashicorp.vercel.app/waypoint/plugins/pack
[/waypoint/plugins/packer]: https://dev-portal-git-zsredirect-waypoint-plugins-hashicorp.vercel.app/waypoint/plugins/packer
[/waypoint/plugins/terraform-cloud]: https://dev-portal-git-zsredirect-waypoint-plugins-hashicorp.vercel.app/waypoint/plugins/terraform-cloud
[/waypoint/plugins/vault]: https://dev-portal-git-zsredirect-waypoint-plugins-hashicorp.vercel.app/waypoint/plugins/vault

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203812579407375

### CODE REVIEW

**Comment by zchsh** [comment]
@BrandonRomano I think this is _nearly_ ready for review & merge, but similar to the comment on 🎟 [Redirect Handling for Legacy Plugin Pages](https://app.asana.com/0/1203792701577283/1203812579407375/f) I thought it might be worth flagging in draft mode first:

- 🧢 There's other work to move integrations in the API from `BrandonRomano` to `HashiCorp`. That'll affect the redirects work here. So I figure might make sense to do that first (in 🎟 [Migrate `BrandonRomano/waypoint` fork to `HashiCorp/waypoint` branch](https://app.asana.com/0/1203792701577283/1203868826798341) I think) and then finalize this PR. Happy to help on that front if there's anything I can do 🙇
- 🚮 Do we want to remove `/waypoint/plugins` as we add these redirects? I assume yes, figured I could roll that into this PR if we want!
-  🔌 We seem to be missing one redirected plugin, we have [/waypoint/plugins/packer](https://developer.hashicorp.com/waypoint/plugins/packer) but not `/waypoint/integrations/BrandonRomano/packer`. My uninformed guess is we will have a `packer` plugin, but we haven't yet extracted into the Integrations API. Let me know if that sounds right, if so, I guess we could not worry about the 404 for now and address it later 👍

I also have some "rewrites" work in #1602 which relates to the above ✍️

File: build-libs/redirects.js [hunk 1]
```diff
@@ -281,6 +281,25 @@ async function buildDevPortalRedirects() {
...
+			source: '/waypoint/plugins',
+			destination: '/waypoint/integrations',
+			permanent: true,
+		},
+		{
+			source: '/waypoint/plugins/:slug',
+			// TODO: should update `BrandonRomano` to `hashicorp` here,
+			// once that change is made in API content
+			destination: '/waypoint/integrations/BrandonRomano/:slug',
+			permanent: true,
```
Comment (BrandonRomano) [inline build-libs/redirects.js:301]: Created a ticket here so we don't forget:

https://app.asana.com/0/1203792701577283/1203879536690930

**Review by BrandonRomano — APPROVED** [review]
@zchsh - this is good!

I actually think we can merge / close out your ticket.  I've created a new ticket to adjust the redirect to include HashiCorp in the slug.

**Comment by BrandonRomano** [comment]
Gonna merge this and then give it a super proper test once it's built on our assembly.

**Pull request merged**